File: test-auth.py
import urllib.request
import json
import ssl

firebaseConfig = {
  "apiKey": "AIzaSyAiXzF-UtDR23AdYzN92ttCFwwkt3KlyAw"
}
url = f"https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key={firebaseConfig['apiKey']}"
data = json.dumps({
    "email": "horsetravel23@gmail.com",
    "password": "Anviphg1",
    "returnSecureToken": True
}).encode('utf-8')

req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

try:
    with urllib.request.urlopen(req, context=ctx) as response:
        print("SUCCESS:", response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print("ERROR:", e.code, e.read().decode('utf-8'))

File: test-auth.js
const firebaseConfig = {
  apiKey: "AIzaSyAiXzF-UtDR23AdYzN92ttCFwwkt3KlyAw",
  authDomain: "antravel.firebaseapp.com",
  projectId: "antravel"
};
// I can just hit the Identity Toolkit API directly with fetch!
const email = "horsetravel23@gmail.com";
const password = "Anviphg1";
const url = `https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=${firebaseConfig.apiKey}`;
fetch(url, {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: JSON.stringify({ email, password, returnSecureToken: true })
}).then(res => res.json()).then(data => console.log(data)).catch(console.error);
